### PR TITLE
feat (apptokens): define/generate protos for service along with some required API definitions/docs 

### PR DESCRIPTION
**Pull request by jimlambrt**
The commits in the PR are intended to show the reviewer the steps need to accomplish the work:

* define new protos
* add protos to Makefile
* generate protobufs
* add new apptoken resources to input.go which is used by `make api`.

### CODE REVIEW

File: internal/proto/controller/api/resources/apptokens/v1/apptoken.proto [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import "controller/api/resources/scopes/v1/scope.proto";
+import "controller/custom_options/v1/options.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+
+option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resources/apptokens;apptokens";
+
+message AppToken {
+  // Output only. The ID of the Account.
```
Comment (uruemu) [inline internal/proto/controller/api/resources/apptokens/v1/apptoken.proto:16]:
```suggestion
  // Output only. The ID of the AppToken.
```

File: internal/api/genapi/input.go [hunk 2]
```diff
@@ -223,6 +224,22 @@ var inputStructs = []*structInfo{
...
+	{
+		inProto:     &apptokens.Grant{},
+		outFile:     "apptokens/grant.gen.go",
+		skipOptions: true,
+	},
+	{
+		inProto:     &apptokens.GrantJson{},
+		outFile:     "apptokens/grant_json.gen.go",
+		skipOptions: true,
+	},
```
Comment (uruemu) [inline internal/api/genapi/input.go:242]: What is the configuration used for?

**Review by uruemu — APPROVED** [review]
LGTM — Left a suggestion to fix a comment and another question.

File: internal/proto/controller/api/resources/apptokens/v1/apptoken.proto (hunk 1 above)
Comment (jimlambrt) [reply internal/proto/controller/api/resources/apptokens/v1/apptoken.proto]: ty!

File: internal/api/genapi/input.go (hunk 2 above)
Comment (jimlambrt) [reply internal/api/genapi/input.go:242]: It's used by internal/api/genapi/main.go via the Makefile in that directory to
generate Go structs which are used by boundary's generated API client to represent
resources when interacting with the API.

**Review by cameronperera — APPROVED** [review]

**Pull request merged**